dashboard/app/controllers/projects_controller.rb

```diff
@@ -133,10 +133,61 @@ def public
     end
   end
 
+  def project_and_featured_project_fields
+    [
+      :storage_apps__id___id,
+      :storage_apps__storage_id___storage_id,
+      :storage_apps__value___value,
+      :storage_apps__project_type___project_type,
+      :storage_apps__published_at___published_at,
+      :featured_projects__featured_at___featured_at,
+      :featured_projects__unfeatured_at___unfeatured_at
+    ]
+  end
+
+  def combine_projects_and_featured_projects_data
+    storage_apps = "pegasus_#{CDO.rack_env}__storage_apps".to_sym
+    project_featured_project_combo_data = DASHBOARD_DB[:featured_projects].select(*project_and_featured_project_fields).
+    join(storage_apps, id: :storage_app_id).map
```

Comment:
this is a confusing use of map, because usually you would pass a method or block to it to transform the dataset. If you want to keep your code roughly the same, you could try replacing `map` will `all` to make it clearer.

However, what I'd actually recommend that you do is take everything inside your `do |project_details|` block below and put it in a new `def extract_data_for_row(project_details)` method, and then call that here like this:
```ruby
@featured_project_table_rows = DASHBOARD_DB[:featured_projects].
  select(...).
  join(...).
  map{|project_details| extract_data_for_row(project_details)}
sort_projects(@featured_project_table_rows)
```

Please feel free to tap me on the shoulder for any help getting this working.